Update code per eslint

It's hard to tell because of other test failures, but `npm run lint:js` now fails due to some new eslint rules and other updates that have come in since then:

```
/Users/ianbicking/src/pageshot/addon/bootstrap.js
   32:15  error  'Services' is not defined               no-undef
   55:5   error  'Services' is not defined               no-undef
   59:5   error  'Services' is not defined               no-undef
   82:5   error  'CustomizableUI' is not defined         no-undef
   83:23  error  'CustomizableUI' is not defined         no-undef
   92:23  error  'CustomizableUI' is not defined         no-undef
   98:5   error  'CustomizableUI' is not defined         no-undef
  142:24  error  'LegacyExtensionsUtils' is not defined  no-undef
  168:24  error  'LegacyExtensionsUtils' is not defined  no-undef
  216:19  error  'AppConstants' is not defined           no-undef
  228:7   error  'Services' is not defined               no-undef
  247:22  error  'PageActions' is not defined            no-undef
  247:53  error  'PageActions' is not defined            no-undef
  247:79  error  'PageActions' is not defined            no-undef

/Users/ianbicking/src/pageshot/addon/webextension/background/analytics.js
  162:11  error  Strings must use doublequote  quotes
  163:22  error  Strings must use doublequote  quotes
  163:43  error  Strings must use doublequote  quotes
  164:20  error  Strings must use doublequote  quotes
  164:39  error  Strings must use doublequote  quotes
  166:17  error  Strings must use doublequote  quotes
  167:17  error  Strings must use doublequote  quotes
  167:36  error  Strings must use doublequote  quotes
  170:11  error  Strings must use doublequote  quotes
  171:22  error  Strings must use doublequote  quotes
  171:43  error  Strings must use doublequote  quotes
  172:20  error  Strings must use doublequote  quotes
  172:39  error  Strings must use doublequote  quotes
  174:17  error  Strings must use doublequote  quotes
  175:17  error  Strings must use doublequote  quotes
  175:36  error  Strings must use doublequote  quotes
  178:11  error  Strings must use doublequote  quotes
  179:22  error  Strings must use doublequote  quotes
  180:20  error  Strings must use doublequote  quotes
  180:39  error  Strings must use doublequote  quotes
  182:17  error  Strings must use doublequote  quotes
  183:17  error  Strings must use doublequote  quotes
  183:36  error  Strings must use doublequote  quotes
  186:11  error  Strings must use doublequote  quotes
  187:22  error  Strings must use doublequote  quotes
  188:20  error  Strings must use doublequote  quotes
  188:39  error  Strings must use doublequote  quotes
  190:17  error  Strings must use doublequote  quotes
  191:17  error  Strings must use doublequote  quotes
  191:36  error  Strings must use doublequote  quotes
  194:11  error  Strings must use doublequote  quotes
  195:22  error  Strings must use doublequote  quotes
  196:20  error  Strings must use doublequote  quotes
  196:39  error  Strings must use doublequote  quotes
  198:17  error  Strings must use doublequote  quotes
  199:17  error  Strings must use doublequote  quotes
  199:36  error  Strings must use doublequote  quotes

/Users/ianbicking/src/pageshot/server/src/pages/shot/controller.js
  18:7  error  'hasSeen' is never reassigned. Use 'const' instead  prefer-const

/Users/ianbicking/src/pageshot/server/src/pages/shot/editor.js
  169:15  error  Expected '===' and instead saw '=='                     eqeqeq
  169:48  error  Expected '===' and instead saw '=='                     eqeqeq
  192:28  error  Strings must use doublequote                            quotes
  208:23  error  Strings must use doublequote                            quotes
  220:26  error  Strings must use doublequote                            quotes
  529:32  error  Expected '===' and instead saw '=='                     eqeqeq
  543:25  error  Expected '===' and instead saw '=='                     eqeqeq
  563:3   error  Expected to return a value at the end of method 'draw'  consistent-return
  568:25  error  Expected '===' and instead saw '=='                     eqeqeq
  568:28  error  Strings must use doublequote                            quotes
  587:32  error  Strings must use doublequote                            quotes
  592:11  error  'rect' is never reassigned. Use 'const' instead         prefer-const
  600:9   error  'rect' is never reassigned. Use 'const' instead         prefer-const
  607:10  error  Unexpected var, use let or const instead                no-var
  608:11  error  'endX' is never reassigned. Use 'const' instead         prefer-const
  609:11  error  'endY' is never reassigned. Use 'const' instead         prefer-const

/Users/ianbicking/src/pageshot/server/src/pages/shot/view.js
  350:9   error  'highlight' is never reassigned. Use 'const' instead  prefer-const
  350:32  error  'highlightEditButton' is missing in props validation  react/prop-types
```

--- a/addon/bootstrap.js
+++ b/addon/bootstrap.js
@@ -1,4 +1,4 @@
-/* globals ADDON_DISABLE */
+/* globals ADDON_DISABLE Services CustomizableUI LegacyExtensionsUtils AppConstants PageActions */
 const ADDON_ID = "screenshots@mozilla.org";
 const TELEMETRY_ENABLED_PREF = "datareporting.healthreport.uploadEnabled";
 const PREF_BRANCH = "extensions.screenshots.";

--- a/addon/webextension/background/analytics.js
+++ b/addon/webextension/background/analytics.js
@@ -159,44 +159,44 @@ this.analytics = (function() {
   // If a cancel event happens between the start and end events, the start time
   // is deleted.
   const rules = [{
-    name: 'page-action',
-    start: { action: 'start-shot', label: 'toolbar-button' },
-    end: { action: 'internal', label: 'unhide-preselection-frame' },
+    name: "page-action",
+    start: { action: "start-shot", label: "toolbar-button" },
+    end: { action: "internal", label: "unhide-preselection-frame" },
     cancel: [
-      { action: 'cancel-shot' },
-      { action: 'internal', label: 'document-hidden' }
+      { action: "cancel-shot" },
+      { action: "internal", label: "document-hidden" }
     ]
   }, {
-    name: 'context-menu',
-    start: { action: 'start-shot', label: 'context-menu' },
-    end: { action: 'internal', label: 'unhide-preselection-frame' },
+    name: "context-menu",
+    start: { action: "start-shot", label: "context-menu" },
+    end: { action: "internal", label: "unhide-preselection-frame" },
     cancel: [
-      { action: 'cancel-shot' },
-      { action: 'internal', label: 'document-hidden' }
+      { action: "cancel-shot" },
+      { action: "internal", label: "document-hidden" }
     ]
   }, {
-    name: 'capture-full-page',
-    start: { action: 'capture-full-page' },
-    end: { action: 'internal', label: 'unhide-preview-frame' },
+    name: "capture-full-page",
+    start: { action: "capture-full-page" },
+    end: { action: "internal", label: "unhide-preview-frame" },
     cancel: [
-      { action: 'cancel-shot' },
-      { action: 'internal', label: 'document-hidden' }
+      { action: "cancel-shot" },
+      { action: "internal", label: "document-hidden" }
     ]
   }, {
-    name: 'capture-visible',
-    start: { action: 'capture-visible' },
-    end: { action: 'internal', label: 'unhide-preview-frame' },
+    name: "capture-visible",
+    start: { action: "capture-visible" },
+    end: { action: "internal", label: "unhide-preview-frame" },
     cancel: [
-      { action: 'cancel-shot' },
-      { action: 'internal', label: 'document-hidden' }
+      { action: "cancel-shot" },
+      { action: "internal", label: "document-hidden" }
     ]
   }, {
-    name: 'make-selection',
-    start: { action: 'make-selection' },
-    end: { action: 'internal', label: 'unhide-selection-frame' },
+    name: "make-selection",
+    start: { action: "make-selection" },
+    end: { action: "internal", label: "unhide-selection-frame" },
     cancel: [
-      { action: 'cancel-shot' },
-      { action: 'internal', label: 'document-hidden' }
+      { action: "cancel-shot" },
+      { action: "internal", label: "document-hidden" }
     ]
   }, {
     name: "save-shot",

--- a/server/src/pages/shot/controller.js
+++ b/server/src/pages/shot/controller.js
@@ -15,7 +15,7 @@ function shouldHighlightEditIcon(model) {
   if (!model.isOwner) {
     return false;
   }
-  let hasSeen = localStorage.hasSeenEditButton;
+  const hasSeen = localStorage.hasSeenEditButton;
   if (!hasSeen && model.enableAnnotations) {
     localStorage.hasSeenEditButton = "1";
   }

--- a/server/src/pages/shot/editor.js
+++ b/server/src/pages/shot/editor.js
@@ -166,7 +166,7 @@ exports.Editor = class Editor extends React.Component {
   }
 
   isColorWhite(color) {
-    if (color == "rgb(255, 255, 255)" || color == "#FFF") {
+    if (color === "rgb(255, 255, 255)" || color === "#FFF") {
       return "white"
     }
     return null;
@@ -189,7 +189,7 @@ exports.Editor = class Editor extends React.Component {
     if (!selectedPos.width || !selectedPos.height) {
       this.removeCropBox();
       this.cropToolBar = null;
-      this.setState({tool: 'pen'});
+      this.setState({tool: "pen"});
       return;
     }
     const x1 = Math.max(selectedPos.left, 0);
@@ -205,7 +205,7 @@ exports.Editor = class Editor extends React.Component {
     croppedContext.drawImage(this.imageCanvas, x1, y1, croppedImage.width, croppedImage.height, 0, 0, croppedImage.width, croppedImage.height);
     const img = new Image();
     const imageContext = this.imageCanvas.getContext("2d");
-    img.crossOrigin = 'Anonymous';
+    img.crossOrigin = "Anonymous";
     const width = cropWidth;
     const height = cropHeight;
     img.onload = () => {
@@ -217,7 +217,7 @@ exports.Editor = class Editor extends React.Component {
     this.canvasHeight = cropHeight;
     this.removeCropBox();
     this.cropToolBar = null;
-    this.setState({tool: 'pen'});
+    this.setState({tool: "pen"});
     sendEvent("confirm-crop", "crop-toolbar");
   }
 
@@ -526,7 +526,7 @@ exports.Editor = class Editor extends React.Component {
       document.addEventListener("mousemove", this.draw);
       document.addEventListener("mousedown", this.setPosition);
       document.addEventListener("mouseup", this.drawMouseup);
-    } else if (this.state.tool == "crop") {
+    } else if (this.state.tool === "crop") {
       document.removeEventListener("mouseup", this.drawMouseup);
       document.removeEventListener("mousemove", this.draw);
       document.removeEventListener("mousedown", this.setPosition);
@@ -540,7 +540,7 @@ exports.Editor = class Editor extends React.Component {
     e.preventDefault();
     drawMousedown = false;
     points = [];
-    if (this.state.tool == "highlighter") {
+    if (this.state.tool === "highlighter") {
       if (this.isColorWhite(this.state.color)) {
         this.imageContext.globalCompositeOperation = "soft-light";
       } else {
@@ -563,9 +563,9 @@ exports.Editor = class Editor extends React.Component {
   draw(e) {
     e.preventDefault();
     if (!drawMousedown) {
-      return null;
+      return;
     }
-    if (this.state.tool == 'highlighter') {
+    if (this.state.tool === "highlighter") {
       this.drawHighlight(e);
     } else {
       this.drawPen(e);
@@ -584,12 +584,12 @@ exports.Editor = class Editor extends React.Component {
   }
 
   drawHighlight(e) {
-    this.drawContext.lineCap = 'square';
+    this.drawContext.lineCap = "square";
     points.push({x: this.pos.x, y: this.pos.y});
     if (points.length < 3) {
       this.drawContext.beginPath();
       this.drawContext.moveTo(this.pos.x, this.pos.y);
-      let rect = this.imageCanvas.getBoundingClientRect();
+      const rect = this.imageCanvas.getBoundingClientRect();
       this.pos.x = e.clientX - rect.left,
       this.pos.y = e.clientY - rect.top
       this.drawContext.lineTo(this.pos.x, this.pos.y);
@@ -597,16 +597,17 @@ exports.Editor = class Editor extends React.Component {
       return;
     }
     this.drawContext.moveTo(this.pos.x, this.pos.y);
-    let rect = this.imageCanvas.getBoundingClientRect();
+    const rect = this.imageCanvas.getBoundingClientRect();
     this.pos.x = e.clientX - rect.left,
     this.pos.y = e.clientY - rect.top
     this.drawContext.lineTo(this.pos.x, this.pos.y);
     this.drawContext.clearRect(0, 0, this.canvasWidth, this.canvasHeight);
     this.drawContext.beginPath();
     this.drawContext.moveTo(points[0].x, points[0].y);
-    for (var i = 1; i < points.length - 2; i++) {
-      let endX = (points[i].x + points[i + 1].x) / 2;
-      let endY = (points[i].y + points[i + 1].y) / 2;
+    let i;
+    for (i = 1; i < points.length - 2; i++) {
+      const endX = (points[i].x + points[i + 1].x) / 2;
+      const endY = (points[i].y + points[i + 1].y) / 2;
       this.drawContext.quadraticCurveTo(points[i].x, points[i].y, endX, endY);
     }
     this.drawContext.quadraticCurveTo(

--- a/server/src/pages/shot/view.js
+++ b/server/src/pages/shot/view.js
@@ -347,7 +347,7 @@ class Body extends React.Component {
 
     let trashOrFlagButton;
     let editButton;
-    let highlight = this.props.highlightEditButton ? <div className="edit-highlight"></div> : null;
+    const highlight = this.props.highlightEditButton ? <div className="edit-highlight"></div> : null;
     if (this.props.isOwner) {
       trashOrFlagButton = <Localized id="shotPageDeleteButton">
         <button className="button transparent trash" title="Delete this shot permanently" onClick={ this.onClickDelete.bind(this) }></button>
@@ -548,6 +548,7 @@ Body.propTypes = {
   downloadUrl: PropTypes.string,
   enableAnnotations: PropTypes.bool,
   expireTime: PropTypes.number,
+  highlightEditButton: PropTypes.bool,
   id: PropTypes.string,
   isExtInstalled: PropTypes.bool,
   isMobile: PropTypes.bool,

